### PR TITLE
base - handle margin mode fix

### PR DESCRIPTION
**Pull request by ttodua**
it does exactly what handleOptionAndParam does. so, we have duplicate codes at this moment.

### CODE REVIEW

**Comment by carlosmiei** [comment]
@ttodua yeah the whole purpose of having `handleOptionAndParam` is to avoid duplicated code like that one

File: js/base/Exchange.js [hunk 1]
```diff
@@ -2793,14 +2793,7 @@ module.exports = class Exchange {
...
          */
-        const defaultMarginMode = this.safeString2 (this.options, 'marginMode', 'defaultMarginMode');
-        const methodOptions = this.safeValue (this.options, methodName, {});
-        const methodMarginMode = this.safeString2 (methodOptions, 'marginMode', 'defaultMarginMode', defaultMarginMode);
-        const marginMode = this.safeStringLower2 (params, 'marginMode', 'defaultMarginMode', methodMarginMode);
-        if (marginMode !== undefined) {
-            params = this.omit (params, [ 'marginMode', 'defaultMarginMode' ]);
-        }
-        return [ marginMode, params ];
+        return this.handleOptionAndParams (params, methodName, 'marginMode');
```
Comment (carlosmiei) [inline js/base/Exchange.js:2796]: @ttodua shouldn't we also accept and route the defaultValue ?

Comment (ttodua) [reply js/base/Exchange.js:2796]: that's what I asked too in chat: https://discord.com/channels/690203284119617602/921046068555313202/1044699473575694437
we don't have any default for it. 
`handleOptionAndParams` does everything as this method did, but neither here we had any default value (should we set `isolated` as default? if no defaults needed, then we might also remove default from `handleSubTypeAndParams`?)

**Comment by carlosmiei** [comment]
@ttodua can you check the build? since we have some exchanges overriding this method, we need to update it there too

**Comment by ttodua** [comment]
> @ttodua can you check the build? since we have some exchanges overriding this method, we need to update it there too

fixed. as a side-thought, I am not aware of the context of creating overrides in derived classes, but if you know the context, can you tell whether we can now (as we have 'default' values available) remove overriden methods at all somehow?

**Comment by carlosmiei** [comment]
> > @ttodua can you check the build? since we have some exchanges overriding this method, we need to update it there too
> 
> fixed. as a side-thought, I am not aware of the context of creating overrides in derived classes, but if you know the context, can you tell whether we can now (as we have 'default' values available) remove overriden methods at all somehow?

@ttodua no it has a different purpose, for some exchanges, we want to apply specific rules. For instance if `swap` default marginMode is cross and if spot `isolated` is the default value because some exchanges only have `cross` for swap and `isolated` for spot/margin

**Pull request merged**